step-functions/step-functions-handler/src/main/java/step/functions/handler/ProcessBasedFunctionHandler.java

```diff
@@ -0,0 +1,268 @@
+            }
+
+            boolean error = false;
+            if (returnCode != 0) {
+                error = true;
+                outputBuilder.appendError("The " + getProcessName() + " execution returned " + returnCode + ". Check logs for more details.");
+            }
+
+            if (error || debug) {
+                attachProcessLogs(process, outputBuilder);
+                takeScreenshot(outputBuilder);
+            }
+
+            if (attachWorkFolder) {
+                attachProcessWorkFolder(process, outputBuilder);
+            }
+
+            return outputBuilder.build();
+        }
+    }
+
+    protected String getSanitizedProcessName() {
+        // Remove leading/trailing whitespace and replace spaces with underscores
+        String sanitized = getProcessName().trim().replaceAll("\\s+", "_");
+        // Use replaceAll() to remove invalid characters
+        sanitized = sanitized.replaceAll("[^a-zA-Z0-9_.]", "_");
+        return sanitized;
+    }
+
+    protected static void createProcessInputAsProperties(ManagedProcess process, Input<?> input) throws IOException {
+        //Create input properties file from keyword properties and keyword arguments
+        Map<String, String> argumentMap = new HashMap<>();
+        Map<String, String> properties = input.getProperties();
+        if (properties != null) {
+            properties.forEach((k, v) -> argumentMap.put(k, v));
+        }
+        // Inputs have precedence over properties
+        JsonObject inputJson = (JsonObject) input.getPayload();
+        if (inputJson != null) {
+            for (String key : inputJson.keySet()) {
+                try {
+                    argumentMap.put(key, inputJson.getString(key));
+                } catch (Exception e) {
+                    throw new RuntimeException("Invalid key/value found for key '" + key + "', only string values are supported.");
+                }
+            }
+        }
+        Properties keywordProperties = new Properties();
+        keywordProperties.putAll(argumentMap);
+        // Write properties to file
+        File stepKeywordPropertiesFile = new File(process.getExecutionDirectory(), KEYWORD_INPUT_PROPERTIES_FILE);
+        try (FileOutputStream outputStream = new FileOutputStream(stepKeywordPropertiesFile)) {
+            keywordProperties.store(outputStream, "Keyword properties");
+        }
+    }
+
+    protected static void createProcessInputMessageAsJson(ManagedProcess process, Input<?> input) throws IOException {
+        //Create input properties file from keyword properties and keyword arguments
+        JsonObjectBuilder outputMessageBuilder = Json.createObjectBuilder();
+        outputMessageBuilder.add("inputs", (JsonObject)  input.getPayload());
+        JsonObjectBuilder propertiesJsonObject = Json.createObjectBuilder();
+        input.getProperties().forEach(propertiesJsonObject::add);
+        outputMessageBuilder.add("properties", propertiesJsonObject);
```

Comment:
Not sure but other attributes from the input object like the Keyword name or timeout might be useful too